### ISSUE TITLE
Deploy on web failed

### ISSUE THREAD

**Issue by Voolu-main** [VapiAI/flutter#1]
Hi Vapi Team,

We've successfully deployed our app using the VAPI API to Android. However, we're encountering compilation errors specifically when targeting the **web build.** Could you please assist us in troubleshooting these issues? 

Errors:

Launching lib\main.dart on Chrome in debug mode...
../../../../AppData/Local/Pub/Cache/hosted/pub.dev/daily_flutter-0.16.0-beta.1/lib/src/platform_bridge.dart:7:8: Error: Dart library 'dart:ffi' is not available on this platform.
import 'dart:ffi';
       ^
Context: The unavailable library 'dart:ffi' is imported through these packages:

    web_entrypoint.dart => package:varch => package:vapi => package:daily_flutter => dart:ffi
    web_entrypoint.dart => package:varch => package:vapi => package:daily_flutter => package:ffi => dart:ffi

Detailed import paths for (some of) the these imports:
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/call_config.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/call_state.dart => package:daily_flutter/src/model/call_config.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/events.dart => package:daily_flutter/src/model/call_state.dart => package:daily_flutter/src/model/call_config.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/events.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:ffi/ffi.dart => package:ffi/src/allocation.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:ffi/ffi.dart => package:ffi/src/arena.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:ffi/ffi.dart => package:ffi/src/utf8.dart => dart:ffi
     web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:ffi/ffi.dart => package:ffi/src/utf16.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/daily_native.dart => package:daily_flutter/src/daily_native.ffigen.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/daily_native.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/daily_native.dart => package:ffi/ffi.dart => package:ffi/src/allocation.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/daily_native.dart => package:ffi/ffi.dart => package:ffi/src/arena.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/daily_native.dart => package:ffi/ffi.dart => package:ffi/src/utf8.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/daily_native.dart => package:ffi/ffi.dart => package:ffi/src/utf16.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/call_config.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/call_state.dart => package:daily_flutter/src/model/call_config.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/events.dart => package:daily_flutter/src/model/call_state.dart => package:daily_flutter/src/model/call_config.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    
     web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/events.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    web_entrypoint.dart => package:varch/main.dart => package:varch/pages/chat_screen.dart => package:vapi/vapi.dart => package:daily_flutter/daily_flutter.dart => package:daily_flutter/src/call_client.dart => package:daily_flutter/src/platform_bridge.dart => package:daily_flutter/src/model.dart => package:daily_flutter/src/model/participants.dart => package:daily_flutter/src/model/common.dart => dart:ffi
    ...

../../../../AppData/Local/Pub/Cache/hosted/pub.dev/daily_flutter-0.16.0-beta.1/lib/src/platform_call_client.dart:3:8: Error: Dart library 'dart:ffi' is not available on this platform.
import 'dart:ffi';
       ^
../../../../AppData/Local/Pub/Cache/hosted/pub.dev/daily_flutter-0.16.0-beta.1/lib/src/texture_renderer.dart:3:8: Error: Dart library 'dart:ffi' is not available on this platform.
import 'dart:ffi';
       ^
../../../../AppData/Local/Pub/Cache/hosted/pub.dev/daily_flutter-0.16.0-beta.1/lib/src/daily_native.dart:3:8: Error: Dart library 'dart:ffi' is not available on this platform.
import 'dart:ffi';
       ^
../../../../AppData/Local/Pub/Cache/hosted/pub.dev/daily_flutter-0.16.0-beta.1/lib/src/model/common.dart:3:8: Error: Dart library 'dart:ffi' is not available on this platform.
import 'dart:ffi';

Unhandled exception:
Unsupported operation: Unsupported invalid type InvalidType(<invalid>) (InvalidType). Encountered while compiling file:///C:/Users/burac/AppData/Local/Pub/Cache/hosted/pub.dev/daily_flutter-0.16.0-beta.1/lib/src/platform_bridge.dart, which contains the type: InvalidType(<invalid>).
#0      ProgramCompiler._typeCompilationError (package:dev_compiler/src/kernel/compiler.dart:3678)
#1      ProgramCompiler._newEmitType (package:dev_compiler/src/kernel/compiler.dart:3471)
#2      ProgramCompiler._emitType (package:dev_compiler/src/kernel/compiler.dart:3379)
#3      ProgramCompiler._emitClassFieldSignature (package:dev_compiler/src/kernel/compiler.dart:1913)
#4      ProgramCompiler._emitClassSignature (package:dev_compiler/src/kernel/compiler.dart:1881)
#5      ProgramCompiler._emitClassDeclaration (package:dev_compiler/src/kernel/compiler.dart:993)
#6      ProgramCompiler._emitClass (package:dev_compiler/src/kernel/compiler.dart:846)
#7      List.forEach (dart:core-patch/growable_array.dart:416)
#8      ProgramCompiler._emitLibrary (package:dev_compiler/src/kernel/compiler.dart:784)
#9      List.forEach (dart:core-patch/growable_array.dart:416)
#10     ProgramCompiler.emitModule (package:dev_compiler/src/kernel/compiler.dart:505)
#11     IncrementalJavaScriptBundler.compile (package:frontend_server/src/javascript_bundle.dart:223)
#12     FrontendCompiler.writeJavaScriptBundle (package:frontend_server/frontend_server.dart:794)
<asynchronous suspension>
#13     FrontendCompiler.compile (package:frontend_server/frontend_server.dart:654)
<asynchronous suspension>
#14     listenAndCompile.<anonymous closure> (package:frontend_server/frontend_server.dart:1303)
<asynchronous suspension>
Failed to compile application.

Out flutter docor -v:

[√] Flutter (Channel stable, 3.19.4, on Microsoft Windows [Version 10.0.22631.3296], locale en-GB)
    • Flutter version 3.19.4 on channel stable at C:\tools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 68bfaea224 (13 days ago), 2024-03-20 15:36:31 -0700
    • Engine revision a5c24f538d
    • Dart version 3.3.2
    • DevTools version 2.31.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0-rc4)
    • Android SDK at C:\Users\burac\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0-rc4
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.4.4)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.4.33213.308
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2021.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] VS Code (version 1.87.2)
    • VS Code at C:\Users\burac\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.86.0

[√] Connected device (4 available)
    • SM A908B (mobile) • RFCMA02W5WT • android-arm64  • Android 12 (API 31)
    • Windows (desktop) • windows     • windows-x64    • Microsoft Windows [Version 10.0.22631.3296]
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 123.0.6312.86
    • Edge (web)        • edge        • web-javascript • Microsoft Edge 123.0.2420.65

[√] Network resources
    • All expected network resources are available.

• No issues found!

Thanks in advance!

**Comment by nikhilro** [comment]
Thanks for reporting. Let me check with the Daily team. For now, does using the Web SDK unblock you?

**Comment by Voolu-main** [comment]
Hi @nikhilro, thanks! No we can't using **Web SDK** in our project!

**Comment by akhilam512** [comment]
Hi @Voolu-main, I'm part of the Daily team. I checked with our engineering team, and unfortunately, it looks like we currently don't support Flutter Web for Daily's Flutter SDK due to all the native integrations that are developed for iOS and Android

**Comment by Voolu-main** [comment]
Hi @akhilam512 
Ok, got it. Thank you very much. Will you also have support for Flutter web in the future? Is this something you are planning?

**Comment by Voolu-main** [comment]
@nikhilro can you check [this issue](https://github.com/VapiAI/flutter/issues/1#issuecomment-2034023411) please? 
Thanks in advance!

**Comment by nikhilro** [comment]
Mind sharing more details on steps to reproduce? Also, create a new issue please

**Comment by Voolu-main** [comment]
@nikhilro ok right now

**Comment by nikhilro** [comment]
(WebRTC requires deeper integration that is only done on Flutter iOS and Android yet)
